src/main/update/check.ts

```diff
@@ -0,0 +1,102 @@
+import { IApv, ISimpleApv } from "src/interfaces/apv";
+import { IDownloadUrls, getDownloadUrls } from "../../utils/url";
+import Headless from "../headless/headless";
+import { get as getConfig, baseUrl, netenv } from "../../config";
+
+export class GetPeersApvFailedError extends Error {}
+
+export interface IUpdate {
+  newApv: ISimpleApv;
+  oldApv: ISimpleApv;
+  urls: IDownloadUrls;
+}
+
+const peerInfos = getConfig("PeerStrings");
+const localApvToken = getConfig("AppProtocolVersion");
+const trustedApvSigners = getConfig("TrustedAppProtocolVersionSigners");
+
+export async function checkForUpdate(
+  standalone: Headless,
+  platform: NodeJS.Platform
+): Promise<IUpdate | null> {
+  let peersApv;
+  try {
+    peersApv = getPeersApv(standalone, peerInfos, trustedApvSigners);
+  } catch (e) {
+    console.error(`getPeersApv Error ocurred ${e}:\n`, e.stderr);
+    return null;
+  }
+
+  const localApv = standalone.apv.analyze(localApvToken);
+
+  if (peersApv.version > localApv.version)
+    return {
+      newApv: peersApv,
+      oldApv: localApv,
+      urls: getDownloadUrls(baseUrl, netenv, peersApv.version, platform),
+    };
+
+  return null;
+}
+
+// FIXME: Could use overload function...
```

Comment:
TypeScript doesn't allow having different bodies for overloads, so probably impossible.

Reply:
😢